core/app/models/spree/line_item.rb

```diff
@@ -19,7 +19,7 @@ class LineItem < Spree::Base
     has_one :product, through: :variant
 
     has_many :adjustments, as: :adjustable, inverse_of: :adjustable, dependent: :destroy
-    has_many :inventory_units, inverse_of: :line_item
+    has_many :inventory_units, inverse_of: :line_item, dependent: false
```

Comment:
Agreed. Maybe worth adding a comment here and point to the `before_destroy :destroy_inventory_units` method?